### PR TITLE
Add tty support for setnsProcess

### PR DESCRIPTION
**Pull request by LK4D4**
I failed to write proper test. Maybe you can help me.

### CODE REVIEW

**Comment by crosbymichael** [comment]
LGTM

**Comment by mrunalp** [comment]
Looking...

File: nsenter/nsexec.c [hunk 1]
```diff
@@ -113,6 +124,28 @@ void nsexec()
...
+			if (dup2(consolefd, STDIN_FILENO) != STDIN_FILENO) {
+				pr_perror("Failed to dup 0");
+				exit(1);
+			}
+			if (dup2(consolefd, STDOUT_FILENO) != STDOUT_FILENO) {
+				pr_perror("Failed to dup 1");
+				exit(1);
+			}
+			if (dup2(consolefd, STDERR_FILENO) != STDERR_FILENO) {
+				pr_perror("Failed to dup 2\n");
```
Comment (mrunalp) [inline nsenter/nsexec.c]: Could take out \n from here for consistency.

**Comment by LK4D4** [comment]
@mrunalp done

**Comment by LK4D4** [comment]
Seems like tests hanged :(

File: nsenter/nsexec.c [hunk 2]
```diff
@@ -113,6 +124,28 @@ void nsexec()
 	}
 
 	if (setjmp(env) == 1) {
+		if (setsid() == -1) {
+			pr_perror("setsid failed");
+			exit(1);
+		}
+		if (consolefd != -1) {
+			if (ioctl(consolefd, TIOCSCTTY, 0) == -1) {
```
Comment (mrunalp) [inline nsenter/nsexec.c]: This is only required on bsd AFAIK.

Comment (mrunalp) [inline nsenter/nsexec.c]: Ignore comment. I see why we do it this way. We need to get the fd before the clone, else the console path isn't accessible.

**Comment by dqminh** [comment]
LGTM

**Comment by mrunalp** [comment]
LGTM

**Pull request merged**